docs/nodes/validate/add-a-validator.md

```diff
@@ -48,182 +48,306 @@ The response has your node’s ID:
+```js
+  const unsignedTx: UnsignedTx = await pchain.buildAddValidatorTx(
+    ...
+    new BN(1e12),
+    ...
+  )
+```
 
-`stakeAmount`
+#### Addresses
 
-In order to validate the Primary Network, one must stake AVAX. This parameter defines the amount of AVAX staked.
+`toAddresses`
 
-`rewardAddress`
+An array of addresses as [Buffer](https://github.com/feross/buffer) who receive the staked tokens at the end of the staking period.
 
-When a validator stops validating the Primary Network, they will receive a reward if they are sufficiently responsive and correct while they validated the Primary Network. These tokens are sent to `rewardAddress`. The original stake will be sent back to an address controlled by `username`.
+`fromAddresses`
+
+An array of addresses as a [Buffer](https://github.com/feross/buffer) who own the staking UTXOs.
+
+`changeAddresses`
+
+Any change resulting from this transaction will be sent to these addresses. You can leave this field empty; if you do, change will be sent to one of the addresses your user controls.
+
+`rewardAddresses`
+
+When a validator stops validating the Primary Network, they will receive a reward if they are sufficiently responsive and correct while they validated the Primary Network. These tokens are sent to `rewardAddresses`. The original stake will be sent back to the addresses defined in `toAddresses`.
 
 A validator’s stake is never slashed, regardless of their behavior; they will always receive their stake back when they’re done validating.
 
-`changeAddr`
+By default, the example uses the variable `pAddressStrings` to define `toAddresses`, `fromAddresses`, `changeAddress` and `rewardAddresses`:
 
-Any change resulting from this transaction will be sent to this address. You can leave this field empty; if you do, change will be sent to one of the addresses your user controls.
+```js
+const pAddressStrings: string[] = pchain.keyChain().getAddressStrings();
+```
 
-`delegationFeeRate`
+This retrieves the P-Chain addresses that belong to the `private key` that appears earlier in the example.
 
-Avalanche allows for delegation of stake. This parameter is the percent fee this validator charges when others delegate stake to them. For example, if `delegationFeeRate` is `1.2345` and someone delegates to this validator, then when the delegation period is over, 1.2345% of the reward goes to the validator and the rest goes to the delegator.
+If you wish to change these addresses, please follow the steps below:
 
-`username` and `password`
+Locate this line in the file
 
-These parameters are the username and password of the user that pays the transaction fee, provides the staked AVAX, and to whom the staked AVAX will be returned.
+```js
+const unsignedTx: UnsignedTx = await pchain.buildAddValidatorTx(
+...
+    pAddressStrings,
+    pAddressStrings,
+    pAddressStrings,
+...
+    pAddressStrings,
```

Comment:
Which one is for which of the following: `toAddresses`, `fromAddresses`, `changeAddress` and `rewardAddresses`. I am totally lost here.

Reply:
Linking to the API signature is critical here so that it is very clear which params is for what.

Reply:
Definitely. Please refer to the comment above regarding `pchain.buldAddValditor.ts`